### PR TITLE
iterables - fixes issue 21863 and now multiset_permutations can handle []

### LINKED ISSUE #21863: multiset_permutations needs to handle []
```diff
diff --git a/sympy/utilities/iterables.py b/sympy/utilities/iterables.py
index 83fc2f48d2..0a91615dde 100644
--- a/sympy/utilities/iterables.py
+++ b/sympy/utilities/iterables.py
@@ -1419,7 +1419,7 @@ def multiset_permutations(m, size=None, g=None):
     do = [gi for gi in g if gi[1] > 0]
     SUM = sum([gi[1] for gi in do])
     if not do or size is not None and (size > SUM or size < 1):
-        if size < 1:
+        if not do and size is None or size < 1:
             yield []
         return
     elif size == 1:
diff --git a/sympy/utilities/tests/test_iterables.py b/sympy/utilities/tests/test_iterables.py
index 221b03f618..b405ac37f5 100644
--- a/sympy/utilities/tests/test_iterables.py
+++ b/sympy/utilities/tests/test_iterables.py
@@ -423,6 +423,9 @@ def test_multiset_permutations():
         [0, 1], [0, 2], [1, 0], [1, 2], [2, 0], [2, 1]]
     assert len(list(multiset_permutations('a', 0))) == 1
     assert len(list(multiset_permutations('a', 3))) == 0
+    for nul in ([], {}, ''):
+        assert list(multiset_permutations(nul)) == [[]], list(multiset_permutations(nul))
+        assert list(multiset_permutations(nul, 1)) == []
 
     def test():
         for i in range(1, 7):
```

### PR DESCRIPTION
**Pull request by praneethratna**
#### Brief description of what is fixed or changed
Fixes #21863 and now multiset_permutations can handle null/empty sets and also added regression tests in test_iterables.py

#### Release Notes

* utilities
    * Fixed bug in iterables.py and now multiset_permutations can handle []

### CODE REVIEW

File: sympy/utilities/iterables.py [hunk 1]
```diff
@@ -1419,7 +1419,7 @@ def multiset_permutations(m, size=None, g=None):
     do = [gi for gi in g if gi[1] > 0]
     SUM = sum([gi[1] for gi in do])
     if not do or size is not None and (size > SUM or size < 1):
-        if size < 1:
+        if not do and size is None or size < 1:
```
Comment (smichr) [inline sympy/utilities/iterables.py:1422]:
```suggestion
        if not do and size is None or size == 0:
```

File: sympy/utilities/tests/test_iterables.py [hunk 2]
```diff
@@ -423,6 +423,9 @@ def test_multiset_permutations():
         [0, 1], [0, 2], [1, 0], [1, 2], [2, 0], [2, 1]]
     assert len(list(multiset_permutations('a', 0))) == 1
     assert len(list(multiset_permutations('a', 3))) == 0
+    for nul in ([], {}, ''):
+        assert list(multiset_permutations(nul)) == [[]], list(multiset_permutations(nul))
+        assert list(multiset_permutations(nul, 1)) == []
```
Comment (smichr) [inline sympy/utilities/tests/test_iterables.py:428]:
```suggestion
        assert list(multiset_permutations(nul)) == [[]]
    assert list(multiset_permutations(nul. 0)) == [[]]
    # impossible requests give no result
    assert list(multiset_permutations(nul, 1)) == []
    assert list(multiset_permutations(nul, -1)) == []
```

**Comment by smichr** [comment]
@praneethratna , please check changes if tests pass

**Comment by praneethratna** [comment]
> @praneethratna , please check changes if tests pass

@smichr everything is fine and this PR can be merged i guess.

**Comment by smichr** [comment]
Thanks, it's in.

**Pull request merged**